src/test/kotlin/com/github/npospolita/worktimesheets/service/CheckinServiceTest.kt

```diff
@@ -0,0 +1,74 @@
+package com.github.npospolita.worktimesheets.service
+
+import com.github.npospolita.worktimesheets.DatabaseTestBase
+import com.github.npospolita.worktimesheets.dao.WorkTimesheetRepository
+import com.github.npospolita.worktimesheets.domain.CheckInType
+import com.github.npospolita.worktimesheets.domain.WorkTimesheetId
+import com.github.npospolita.worktimesheets.domain.errors.ValidationError
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.assertThrows
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.test.annotation.DirtiesContext
+import java.time.LocalDate
+import java.time.ZoneId
+import javax.transaction.Transactional
+
+@SpringBootTest
+@Transactional
+class CheckInServiceTest(
+    @Autowired val checkInService: CheckInService,
+    @Autowired val workTimesheetRepository: WorkTimesheetRepository
+) : DatabaseTestBase() {
+
+    @Test
+    fun new_CheckIn() {
+        checkInService.checkIn(CheckInType.IN, 1)
+
+        val timesheetRecord =
+            workTimesheetRepository.findByIdOrNull(WorkTimesheetId(LocalDate.now(ZoneId.of("Europe/Moscow")), 1))
+        assertNotNull(timesheetRecord)
+        assertNotNull(timesheetRecord!!.startTime)
+    }
+
+    @Test
+    fun new_CheckOut() {
+        assertThrows<ValidationError> {
+            checkInService.checkIn(CheckInType.OUT, 1)
+        }
+    }
+
+    @Test
+    fun existed_CheckInTwice() {
```

Comment:
Тесты можно писать так:
```suggestion
    fun `existed check in twice`() {
```